Sources/SwifterSwift/Foundation/NSRegularExpressionExtensions.swift

```diff
@@ -0,0 +1,146 @@
+    func matches(in string: String,
+                 options: MatchingOptions = [],
+                 range: Range<String.Index>) -> [NSTextCheckingResult] {
+        return matches(in: string,
+                       options: options,
+                       range: NSRange(range, in: string))
+    }
+
+    /// Returns the number of matches of the regular expression within the specified range of the string.
+    ///
+    /// - Parameters:
+    ///   - string: The string to search.
+    ///   - options: The matching options to use. See NSRegularExpression.MatchingOptions for possible values.
+    ///   - range: The range of the string to search.
+    /// - Returns: The number of matches of the regular expression.
+    func numberOfMatches(in string: String,
+                         options: MatchingOptions = [],
+                         range: Range<String.Index>) -> Int {
+        return numberOfMatches(in: string,
+                               options: options,
+                               range: NSRange(range, in: string))
+    }
+
+    /// Returns the first match of the regular expression within the specified range of the string.
+    ///
+    /// - Parameters:
+    ///   - string: The string to search.
+    ///   - options: The matching options to use. See `NSRegularExpression.MatchingOptions` for possible values.
+    ///   - range: The range of the string to search.
+    /// - Returns: An `NSTextCheckingResult` object. This result gives the overall matched range via its `range` property, and the range of each individual capture group via its `range(at:)` method. The range {NSNotFound, 0} is returned if one of the capture groups did not participate in this particular match.
+    func firstMatch(in string: String,
+                    options: MatchingOptions = [],
+                    range: Range<String.Index>) -> NSTextCheckingResult? {
+        return firstMatch(in: string,
+                          options: options,
+                          range: NSRange(range, in: string))
+    }
+
+    /// Returns the range of the first match of the regular expression within the specified range of the string.
+    ///
+    /// - Parameters:
+    ///   - string: The string to search.
+    ///   - options: The matching options to use. See `NSRegularExpression.MatchingOptions` for possible values.
+    ///   - range: The range of the string to search.
+    /// - Returns: The range of the first match. Returns `nil` if no match is found.
+    func rangeOfFirstMatch(in string: String,
+                           options: MatchingOptions = [],
+                           range: Range<String.Index>) -> Range<String.Index>? {
+        return Range(rangeOfFirstMatch(in: string,
```

Comment:
If we are always creating a range and returning this will never be nil right?

Reply:
This will be `nil` if the regex can't be found in the string. It's why I added [this test](https://github.com/SwifterSwift/SwifterSwift/pull/727/commits/f2750fd58c74bbbef864a255a56c2f90a70a9701).

Reply:
Nice :))